gptfixer/gpt.c

```diff
@@ -0,0 +1,786 @@
+            header->DiskGUID.guid[5], header->DiskGUID.guid[6], header->DiskGUID.guid[7], header->DiskGUID.guid[8],
+            header->DiskGUID.guid[9], header->DiskGUID.guid[10], header->DiskGUID.guid[11], header->DiskGUID.guid[12],
+            header->DiskGUID.guid[13], header->DiskGUID.guid[14], header->DiskGUID.guid[15], header->PartitionEntryLBA,
+            header->NumberOfPartitionEntries, header->SizeOfPartitionEntry, header->PartitionEntryArrayCRC32);
+    for_each_gpt_entry (i, size, count, gpt, entry) {
+        if (unused_entry(entry)) {
+            continue;
+        }
+        fprintf(
+            stdout,
+            "GPT Partition Table Entry:\n"
+            "PartitionTypeGUID:        "
+            "%02x%02x%02x%02x-%02x%02x-%02x%02x-%02x%02x-%02x%02x%"
+            "02x%02x%02x%"
+            "02x\n"
+            "UniquePartitionGUID:      "
+            "%02x%02x%02x%02x-%02x%02x-%02x%02x-%02x%02x-%02x%02x%"
+            "02x%02x%02x%"
+            "02x\n"
+            "StartingLBA:              0x%" PRIx64 "\n"
+            "EndingLBA:                0x%" PRIx64 "\n"
+            "Attributes:               0x%016" PRIx64 "\n",
+            entry->PartitionTypeGUID.guid[0], entry->PartitionTypeGUID.guid[1], entry->PartitionTypeGUID.guid[2],
+            entry->PartitionTypeGUID.guid[3], entry->PartitionTypeGUID.guid[4], entry->PartitionTypeGUID.guid[5],
+            entry->PartitionTypeGUID.guid[6], entry->PartitionTypeGUID.guid[7], entry->PartitionTypeGUID.guid[8],
+            entry->PartitionTypeGUID.guid[9], entry->PartitionTypeGUID.guid[10], entry->PartitionTypeGUID.guid[11],
+            entry->PartitionTypeGUID.guid[12], entry->PartitionTypeGUID.guid[13], entry->PartitionTypeGUID.guid[14],
+            entry->PartitionTypeGUID.guid[15], entry->UniquePartitionGUID.guid[0], entry->UniquePartitionGUID.guid[1],
+            entry->UniquePartitionGUID.guid[2], entry->UniquePartitionGUID.guid[3], entry->UniquePartitionGUID.guid[4],
+            entry->UniquePartitionGUID.guid[5], entry->UniquePartitionGUID.guid[6], entry->UniquePartitionGUID.guid[7],
+            entry->UniquePartitionGUID.guid[8], entry->UniquePartitionGUID.guid[9], entry->UniquePartitionGUID.guid[10],
+            entry->UniquePartitionGUID.guid[11], entry->UniquePartitionGUID.guid[12],
+            entry->UniquePartitionGUID.guid[13], entry->UniquePartitionGUID.guid[14],
+            entry->UniquePartitionGUID.guid[15], entry->StartingLBA, entry->EndingLBA, entry->Attributes);
+    }
+}
+
+static void sync_pwrite(int fd, const void *data, uint64_t length, uint64_t offset) {
+    CHECK(fd >= 0);
+    CHECK(data != NULL);
+    CHECK(length <= SSIZE_MAX);
+    CHECK((off_t)offset >= 0);
+    CHECK((uint64_t)(off_t)offset == offset);
+
+    ssize_t res = pwrite64(fd, data, (size_t)length, (off_t)offset);
+    CHECK(res >= -1);
+    if (res == -1)
+        err(1, "write error");
+    CHECK((size_t)res <= length);
+    if ((size_t)res != length)
```

Comment:
Those two CHECK actually make the code harder to follow. `rest == -1` and `res != length` already handle the actual error cases. Asserts for kernel/glibc code (as those CHECK are) make one think if there aren't any real error cases here